src/vs/workbench/contrib/terminalContrib/suggest/browser/terminal.suggest.contribution.ts

```diff
@@ -351,11 +352,16 @@ registerActiveInstanceAction({
 	title: localize2('workbench.action.terminal.hideSuggestWidget', 'Hide Suggest Widget'),
 	f1: false,
 	precondition: ContextKeyExpr.and(ContextKeyExpr.or(TerminalContextKeys.processSupported, TerminalContextKeys.terminalHasBeenCreated), TerminalContextKeys.focus, TerminalContextKeys.isOpen, TerminalContextKeys.suggestWidgetVisible),
-	keybinding: {
+	keybinding: [{
 		primary: KeyCode.Escape,
 		// Escape is bound to other workbench keybindings that this needs to beat
 		weight: KeybindingWeight.WorkbenchContrib + 1
 	},
+	{
+		primary: KeyCode.UpArrow,
+		when: ContextKeyExpr.and(SimpleSuggestContext.FocusedFirstSuggestion, ContextKeyExpr.equals(`config.${TerminalSuggestSettingId.UpArrowNavigatesHistory}`, true)),
+		weight: KeybindingWeight.WorkbenchContrib + 2
+	}],
```

Comment:
I don't think this will pass the up arrow keypress through when it's hidden? We want it to start navigating history on first keypress, not second.

Reply:
yes, good point. instead of resuing this one, i'll add a new action that hides and sends up arrow to the terminal.